### DISCUSSION TITLE
Question: How can I set up a filter that, when I select a specific term, allows me to filter by that term as well as all the subterm within it

### DISCUSSION THREAD

**Discussion by Harshagracy** [microsoft-search/pnp-modern-search#3269] [Q&A] [answered]
Question: How can I set up a filter that, when I select a specific term, allows me to filter by that term as well as all the subterm within it for more granular data analysis?
Screenshot of the termset :
<img width="131" alt="image" src="https://github.com/microsoft-search/pnp-modern-search/assets/14230498/504d43ef-4c3e-4374-8720-8d9f13dabcf1">
Screenshot of Refinable string:
<img width="726" alt="image" src="https://github.com/microsoft-search/pnp-modern-search/assets/14230498/8bd817b3-68a9-4a3b-918e-e0f9adcfb42f">
Filter Configuration: 
<img width="390" alt="image" src="https://github.com/microsoft-search/pnp-modern-search/assets/14230498/9f305aca-15f5-41f6-a2c0-38a64baf9962">
<img width="535" alt="image" src="https://github.com/microsoft-search/pnp-modern-search/assets/14230498/acda981d-c345-46fb-b65c-5c7b89222038">
<img width="534" alt="image" src="https://github.com/microsoft-search/pnp-modern-search/assets/14230498/0326e8c5-075c-4f09-89fa-0d04f04aa054">
As you can see if i filter using the second filter, having "GPP" as the prefix, it successfully retrieves the states within a country. However, my objective is to include the country itself in the results as well. Is it feasible to achieve this, and if so, what configuration adjustments should be made to accomplish it?

**Comment by wobba** [comment]
See samples at the bottom of my post https://www.techmikael.com/2014/03/s15e01-kql-basics.html

**Reply by Harshagracy** [reply]
Hi @wobba,  Thank you so much for your response. The example you provided in the article seems like it would address my question:
```
All restaurants in India, including India itself
owstaxIdLocation:"GPP|#c8a43f13-5ea1-45f2-b46d-3a1986a1cbd7" AND owstaxIdLocation:"GP0|#c8a43f13-5ea1-45f2-b46d-3a1986a1cbd7"
```
The AND condition didnt work for me but the OR worked:

`[location2="GP0|#f3c2a858-64df-44c4-a05c-b1139e2948b3"OR location2="GPP|#f3c2a858-64df-44c4-a05c-b1139e2948b3"]`
<img width="586" alt="image" src="https://github.com/microsoft-search/pnp-modern-search/assets/14230498/ec6157d8-67e0-4d10-8fab-1958a8cf0760">

I have two questions:

- How can I configure this in the PnP search filter in a user-friendly format, displaying the country title?
- How can I make it a single option? Is it possible?

**Reply by wobba** [reply, marked as answer]
@Harshagracy an error in the the post.. OR makes sense indeed :) And.. unfortunately there are no automatic ways to build this today. You can try a custom filter template and see if you re-format the token value to what you need.

**Reply by Harshagracy** [reply]
I appreciate it, @wobba! I also had the same impression that it might not be achievable through the out of the box options. I'll give it a shot with custom filter templates. Thanks once more!
